src/nccl_ofi_scheduler.c

```diff
@@ -53,6 +53,50 @@ void nccl_net_ofi_set_multiplexing_schedule(size_t size, int num_rails,
 	}
 }
 
+void nccl_net_ofi_set_multiplexed_round_robin_schedule(nccl_net_ofi_threshold_scheduler_t *scheduler,
+												   size_t size,
+												   int num_rails,
+					    						   size_t align,
+					    						   nccl_net_ofi_schedule_t *schedule)
+{
+	int rail_id;
+	int rail_group_size = 2;
+	nccl_net_ofi_mutex_lock(&scheduler->rr_lock);
+
+	/* Retieve and increment multiplex-round-robin counter; wrap around if required */
+	rail_id = scheduler->mux_rr_counter;
+	scheduler->mux_rr_counter += rail_group_size;
+	scheduler->mux_rr_counter = scheduler->mux_rr_counter == num_rails ? 0 : scheduler->mux_rr_counter;
+
+	nccl_net_ofi_mutex_unlock(&scheduler->rr_lock);
+
+	/* Number of bytes left to assign */
+	size_t left = size;
+	/* Offset into message */
+	size_t offset = 0;
+	/* Maximum size of a stripe */
+	size_t max_stripe_size = 0;
+
+	schedule->num_xfer_infos = 0;
+
+	if (OFI_UNLIKELY(num_rails == 0)) return;
```

Comment:
Parameter checks are better done at the beginning of the function, before you start modifying things.

Reply:
Done